src/SFML/Window/WindowImpl.cpp

```diff
@@ -277,9 +273,10 @@ void WindowImpl::processJoystickEvents()
                 if (prevPressed ^ currPressed)
                 {
                     Event event;
-                    event.type = currPressed ? Event::JoystickButtonPressed : Event::JoystickButtonReleased;
-                    event.joystickButton.joystickId = i;
-                    event.joystickButton.button     = j;
+                    if (currPressed)
+                        event = Event::JoystickButtonPressed{i, j};
+                    else
+                        event = Event::JoystickButtonReleased{i, j};
```

Comment:
How about:

```cpp
if (currPressed)
    pushEvent(Event::JoystickButtonPressed{i, j});
else
    pushEvent(Event::JoystickButtonReleased{i, j});
```

?

Reply:
@ChrisThrasher: you put thumbs-up and resolved the comment, but I don't see the suggestion applied. Did you change your mind?